src/app/components/SearchForm/WithSearch.jsx

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/label-has-for */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getQueryString } from '../../search/query';
+import { initialSearchQuery, searchQueryPropTypes } from '../../stores/InitialState';
+import { errorMessagesText } from '../../constants/labels';
+
+function withSearch(WrappedComponent) {
+  class SearchComponent extends React.Component {
+    constructor(props) {
+      super(props);
+
+      this.state = { ...props, ...{ error: false, errorMsg: '', isFeatureFlagsActivated: {} } };
+
+      this.onFieldChange = this.onFieldChange.bind(this);
+      this.onQueryChange = this.onQueryChange.bind(this);
+      this.handleSubmit = this.handleSubmit.bind(this);
+      this.submitSearchRequest = this.submitSearchRequest.bind(this);
+    }
+
+    componentDidMount() {
+      global.window.scrollTo(0, 0);
+    }
+
+    onFieldChange(event) {
+      const fieldSelected = event.target.value;
+      this.setState((prevState) => {
+        const advancedQuery = {
+          query: prevState.searchQuery.showQuery
+            ? prevState.searchQuery.showQuery : prevState.searchQuery.queries[0].query,
+          field: fieldSelected,
+        };
+        return ({
+          searchQuery: Object.assign({}, initialSearchQuery,
+            { showField: '', showQuery: '' },
+            { queries: [].concat(advancedQuery) }),
+        });
+      });
+    }
+
+    onQueryChange(event) {
+      const querySelected = event.target.value;
+
+      this.setState((prevState) => {
+        const advancedQuery = {
+          query: querySelected,
+          field: prevState.searchQuery.showField ? prevState.searchQuery.showField : prevState.searchQuery.queries[0].field || 'keyword',
+        };
+
+        return ({
+          searchQuery: Object.assign({}, initialSearchQuery, { showField: '', showQuery: querySelected },
+            { queries: [].concat(advancedQuery) }),
+        });
+      });
+      if (querySelected) {
+        this.setState({ error: false, errorMsg: '' });
+      }
+    }
+
+    handleSubmit(event) {
+      if (event && event.charCode === 13) {
+        this.props.userQuery(
+          Object.assign({}, initialSearchQuery, {
+            query: this.state.searchQuery.queries,
+          }),
+        );
+      }
+    }
+
+    submitSearchRequest(event) {
+      event.preventDefault();
+      const query = this.state.searchQuery.queries[0].query.replace(/^\s+/, '').replace(/\s+$/, '');
```

Comment:
This method for removing leading/trailing whitespace works, but I think `trim()` is shorter and more understandable to anyone who isn't familiar with regexes?